plugins/in_winevtlog/winevtlog.c

```diff
@@ -0,0 +1,792 @@
+{
+    PEVT_VARIANT values;
+    DWORD buffer_size = 0;
+    DWORD buffer_size_used = 0;
+    DWORD count = 0;
+    BOOL succeeded = FLB_TRUE;
+
+    EVT_HANDLE context = EvtCreateRenderContext(0, NULL, EvtRenderContextUser);
+    if (context == NULL) {
+        flb_error("Failed to create renderContext");
+        succeeded = FLB_FALSE;
+        goto cleanup;
+    }
+
+    // Get the size of the buffer
+    EvtRender(context, handle, EvtRenderEventValues, 0, NULL, &buffer_size, &count);
+    values = (PEVT_VARIANT)flb_malloc(buffer_size);
+
+    succeeded = EvtRender(context,
+                          handle,
+                          EvtRenderContextValues,
+                          buffer_size,
+                          values,
+                          &buffer_size_used,
+                          &count);
+
+    if (!succeeded) {
+        flb_error("Failed to get string inserts with %d\n", GetLastError());
+        goto cleanup;
+    }
+
+    *prop_count = count;
+    *string_inserts_values = values;
+    *string_inserts_size = buffer_size;
+
+cleanup:
+
+    if (context != NULL)
+        EvtClose(context);
+
+    return succeeded;
+}
+
+static int winevtlog_next(struct winevtlog_channel *ch, int hit_threshold)
+{
+    EVT_HANDLE events[SUBSCRIBE_ARRAY_SIZE];
+    DWORD count = 0;
+    DWORD status = ERROR_SUCCESS;
+    BOOL has_next = FALSE;
+
+    /* If subscription handle is NULL, it should return false. */
+    if (!ch->subscription) {
+        flb_error("Invalid subscription is passed");
+        return FLB_FALSE;
+    }
+
+    if (hit_threshold) {
+        return FLB_FALSE;
+    }
+
+    has_next = EvtNext(ch->subscription, SUBSCRIBE_ARRAY_SIZE,
+                       events, INFINITE, 0, &count);
+
+    if (!has_next) {
+        status = GetLastError();
+        if (ERROR_CANCELLED == status) {
+            return FLB_FALSE;
+        }
+        if (ERROR_NO_MORE_ITEMS != status) {
+            return FLB_FALSE;
+        }
+    }
+
+    if (status == ERROR_SUCCESS) {
+        ch->count = count;
+        for (int i = 0; i < count; i++) {
+            ch->events[i] = events[i];
+            EvtUpdateBookmark(ch->bookmark, ch->events[i]);
+        }
+
+        return FLB_TRUE;
+    }
+
+    return FLB_FALSE;
+}
+
+/*
+ * Read from an open Windows Event Log channel.
+ */
+int winevtlog_read(struct winevtlog_channel *ch, msgpack_packer *mp_pck, struct winevtlog_config *ctx,
+                   unsigned int *read)
+{
+    DWORD status = ERROR_SUCCESS;
+    PWSTR system_xml = NULL;
+    unsigned int system_size = 0;
+    unsigned int message_size = 0;
+    unsigned int string_inserts_size = 0;
+    int hit_threshold = FLB_FALSE;
+    unsigned int read_size = 0;
+    PWSTR message = NULL;
+    PEVT_VARIANT rendered_system = NULL;
+    PEVT_VARIANT string_inserts = NULL;
+    UINT count_inserts = 0;
+    DWORD i = 0;
+
+    while (winevtlog_next(ch, hit_threshold)) {
+        for (DWORD i = 0; i < ch->count; i++) {
```

Comment:
I think you forgot to remove this definition.

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/4179/commits/f18098bc507d377ad66b547db1e984929fd66a75.

Reply:
Yeah. Actually, I forgot to remove variable definition here. Thanks for the pointing out.